ams/server/src/main/java/com/netease/arctic/server/manager/SparkOptimizerContainer.java

```diff
@@ -0,0 +1,448 @@
+                .build();
+        String namespace =
+            StringUtils.defaultIfEmpty(
+                sparkConf.configValue(SparkConfKeys.KUBERNETES_NAMESPACE), "default");
+        startUpStatesMap.put(
+            KUBERNETES_SUBMISSION_ID_PROPERTY,
+            String.format("%s:%s", namespace, kubernetesDriverName(resource)));
+      } else {
+        String applicationId = fetchCommandOutput(exec, yarnApplicationIdReader);
+        if (applicationId != null) {
+          startUpStatesMap.put(YARN_APPLICATION_ID_PROPERTY, applicationId);
+        }
+      }
+      return startUpStatesMap;
+    } catch (IOException e) {
+      throw new UncheckedIOException("Failed to scale out spark optimizer.", e);
+    }
+  }
+
+  @Override
+  protected String buildOptimizerStartupArgsString(Resource resource) {
+    Map<String, String> sparkConfig = loadSparkConfig();
+    SparkOptimizerContainer.SparkConf resourceSparkConf =
+        SparkOptimizerContainer.SparkConf.buildFor(sparkConfig, getContainerProperties())
+            .withGroupProperties(resource.getProperties())
+            .build();
+
+    // Default enable the spark DRA(dynamic resource allocation)
+    resourceSparkConf.putToOptions(
+        SparkConfKeys.KUBERNETES_DRA_ENABLED,
+        StringUtils.defaultIfEmpty(
+            resourceSparkConf.configValue(SparkConfKeys.KUBERNETES_DRA_ENABLED), "true"));
+    resourceSparkConf.putToOptions(
+        SparkConfKeys.KUBERNETES_DRA_MAX_EXECUTORS,
+        StringUtils.defaultIfEmpty(
+            resourceSparkConf.configValue(SparkConfKeys.KUBERNETES_DRA_MAX_EXECUTORS),
+            String.valueOf(resource.getThreadCount())));
+
+    if (deployedOnKubernetes()) {
+      addKubernetesProperties(resource, resourceSparkConf);
+    } else {
+      addYarnProperties(resourceSparkConf);
+    }
+    String sparkOptions = resourceSparkConf.toConfOptions();
+    String proxyUser =
+        getContainerProperties()
+            .getOrDefault(
+                OptimizerProperties.EXPORT_PROPERTY_PREFIX + ENV_HADOOP_USER_NAME, "hadoop");
+    String jobArgs = super.buildOptimizerStartupArgsString(resource);
+    // ./bin/spark-submit --master <master> --deploy-mode=<sparkMode> <options> --proxy-user <user>
+    // --class
+    // <main-class>
+    // <job-file>
+    // <arguments>
+    //  options: --conf <property=value>
+    return String.format(
+        "%s/bin/spark-submit --master %s --deploy-mode=%s %s --proxy-user %s --class %s %s %s",
+        sparkHome,
+        sparkMaster,
+        deployMode.getValue(),
+        sparkOptions,
+        proxyUser,
+        SPARK_JOB_MAIN_CLASS,
+        jobUri,
+        jobArgs);
+  }
+
+  private Map<String, String> loadSparkConfig() {
+    try {
+      return Arrays.stream(new org.apache.spark.SparkConf().getAll())
+          .collect(Collectors.toMap(Tuple2::_1, Tuple2::_2));
+    } catch (Exception e) {
+      LOG.error("load spark conf failed: {}", e.getMessage());
```

Comment:
```suggestion
      LOG.error("Load spark conf failed.", e);
```

Reply:
@zhoujinsong Thx for your comments. It's really looks good to me. Fixed